apps/login/app/(login)/password/page.tsx

```diff
@@ -1,32 +1,48 @@
-"use client";
-import { Button, ButtonVariants } from "#/ui/Button";
-import { TextInput } from "#/ui/Input";
+import { getSession, server } from "#/lib/zitadel";
+import Alert from "#/ui/Alert";
+import PasswordForm from "#/ui/PasswordForm";
 import UserAvatar from "#/ui/UserAvatar";
-import { useRouter } from "next/navigation";
+import { getMostRecentCookieWithLoginname } from "#/utils/cookies";
 
-export default function Page() {
-  const router = useRouter();
+export default async function Page({ searchParams }: { searchParams: any }) {
+  const { loginName } = searchParams;
+  const sessionFactors = await loadSession(loginName);
+
+  async function loadSession(loginName: string) {
+    try {
+      const recent = await getMostRecentCookieWithLoginname(loginName);
+
+      return getSession(server, recent.id, recent.token).then(({ session }) => {
+        return session;
+      });
+    } catch (error) {
+      console.log(error);
+    }
+  }
```

Comment:
a client error is thrown now 👍

Reply:
I'd keep this as it is, as moving this to a seperate file does not improve code quality IMO. This should be visible as server side executed code. If it will repeat itself further we should come up with a service for it